### PR TITLE
Fixes #132: option added to push documentation to GH-PAGES

### LINKED ISSUE #132: add an option to WEBUI for pushing documentation to GH-PAGES
### This is a
- [ ] Bug report
- [x] Feature request

### Current behavior

No option to push documentation 
### Expected behavior

once documentation is generated, ask the user that whether they want to push documentation or not 

### Steps to reproduce
1.
2.
3.

### PR DESCRIPTION
**Pull request by sch00lb0y**
## Description

option added to push documentation to GH-PAGES
## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96 please check the implementation.

**Comment by pri22296** [comment]
@sch00lb0y please fix codacy issues

File: ghdeploy.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
```
Comment (pri22296) [inline ghdeploy.sh]: Instead of creating this script, use the existing `publish_docs.sh` otherwise it would again lead to situations like #108 . ensure that the script is usable from both a webUI and travis. use idea similar to `generate.sh` to run conditional code based on whether running from a webUI.

**Comment by imujjwal96** [comment]
@sch00lb0y  Can we see a deployment for this?

**Comment by sch00lb0y** [comment]
@imujjwal96 [demo link](http://pagupu.in)

File: ghdeploy.sh (hunk 1 above)
Comment (sch00lb0y) [inline ghdeploy.sh]: I have made changes

File: publish_docs.sh [hunk 2]
```diff
@@ -32,19 +53,32 @@ fi
...
+else
+  cp -a ../_build/html/. ./
+fi
 
 # Enable publishing documentation to custom URL
-if [[ -z "${DOCURL}" ]]; then
-  echo -e "DOCURL not set. Using default github pages URL"
-else
-  echo $DOCURL > CNAME
+if ![ "${WEBUI:-false}" == "true" ]; then
```
Comment (pri22296) [inline publish_docs.sh]: why was this change made?

File: backend/generator.js [hunk 3]
```diff
@@ -41,10 +39,42 @@ exports.executeScript = function (socket, formData) {
...
   });
   return true;
-};
\ No newline at end of file
+};
+
+exports.deployPages = function (socket, data) {
+  var donePercent = 0;
+  var projectName = data.gitURL.split("/")[4].split(".")[0];
+  var gitURL = "http://" + data.username+":" + crypter.decrypt(data.encryptedToken) + "@github.com/" + data.username + "/" + projectName;
```
Comment (pri22296) [inline backend/generator.js]: Instead of building the url here, you could have passed the token as OAUTH_TOKEN and the publish script should have handled that.

File: publish_docs.sh (hunk 2 above)
Comment (sch00lb0y) [inline publish_docs.sh]: now WebUI don't have CNAME. I'll remove the condition

**Comment by sch00lb0y** [comment]
@pri22296 for pushing repo have to get the access.even in open-event-webapp if users want to push, they requesting delete level access

File: publish_docs.sh (hunk 2 above)
Comment (pri22296) [inline publish_docs.sh]: yeah but users can still specify the docurl in the .yaydoc.yml file.

**Comment by pri22296** [comment]
@sch00lb0y If that's the case, then let it be

File: backend/generator.js [hunk 4]
```diff
@@ -1,15 +1,13 @@
-var exports = module.exports = {};
```
Comment (imujjwal96) [inline backend/generator.js]: why did you remove this?

File: backend/generator.js [hunk 5]
```diff
@@ -41,10 +39,45 @@ exports.executeScript = function (socket, formData) {
...
     } else {
       socket.emit('failure', {errorCode: code});
     }
   });
   return true;
-};
\ No newline at end of file
+};
+
+exports.deployPages = function (socket, data) {
```
Comment (imujjwal96) [inline backend/generator.js]: `generator.js` is the backend script for documentation generation. Move this function to a separate file.

File: public/scripts/form.js [hunk 6]
```diff
@@ -44,6 +44,7 @@ $(function () {
       $('#notification-container').css("visibility", "hidden");
       $('#notification-container').css("opacity", "0");
     }, 5000)
+    $("#buttons > .row").append('<a href="/github?email='+data.email+'&uniqueId='+data.uniqueId+'&gitURL='+data.gitUrl+'" class="btn btn-default" id="btnDeploy">deploy</a>')
```
Comment (imujjwal96) [inline public/scripts/form.js]: Please follow the same pattern as above

**Comment by imujjwal96** [comment]
The link is not working

**Comment by sch00lb0y** [comment]
@pri22296 @imujjwal96  I have made changes. here is the new link http://yaydoc.pagupu.in please check

**Comment by imujjwal96** [comment]
Although the documentation was pushed successfully, the progress bar is not completed.
image

**Comment by sch00lb0y** [comment]
@imujjwal96  I have fixed please check now

**Comment by imujjwal96** [comment]
The reason why webapp requires `Delete` permission

image

**Comment by sch00lb0y** [comment]
@imujjwal96 i have updated the permission scope

File: views/index.jade [hunk 7]
```diff
@@ -35,6 +35,7 @@ block content
             .row
               a.btn.btn-default(type='button' id='btnDownload') Download
               a.btn.btn-default(type='button' id='btnPreview' target='_blank') Preview
+              a.btn.btn-default(type='button' id='btnDeploy') deploy
```
Comment (imujjwal96) [inline views/index.jade]: **D**eploy :)

**Comment by pri22296** [comment]
any specific reason for not reusing the existing console for ghpages deploy?

**Comment by sch00lb0y** [comment]
@pri22296 If we keep on adding things to one page. it'll be hard to maintain and also I'm templating the deploy page according to the need. In `open-event-webapp` also they maintaining separate page for deploying
